testplan/testing/multitest/driver/http/server.py

```diff
@@ -14,27 +15,124 @@
 
 from ..base import Driver, DriverConfig
 
+_CONTENT_TYPE_KEY = "Content-Type"
+_CONTENT_LENGTH_KEY = "Content-Length"
+
+
+class ReceivedRequest:
+    """
+    Stores information of requests received by the HTTP Server.
+    """
+
+    def __init__(
+        self,
+        method: str,
+        path_url: str,
+        headers: dict,
+        raw_data: bytes,
+        raw_requestline: bytes,
+        requestline: str,
+        request_version: str,
+    ) -> None:
+        """
+        Constructs a new received request
+
+        :param method: method of the request, eg.: GET, POST, HEAD...
+        :param path_url: URL of the request
+        :param headers: Dictionary of HTTP Headers sent with the request
+        :param raw_data: Data sent in the body of the request in original bytes format
+        :param raw_requestline: The first line of the request containing
+            the method, URL and version in original bytes format
+        :param requestline: The first line of the request containing
+            the method, URL and version in string format
+        :param request_version: Human-readable string of the protocol version
+        """
+
+        # Default empty dict for headers.
+        headers = {} if headers is None else headers
+
+        self.method = method
+        self.path_url = path_url
+        self.headers = headers
+        self.raw_data = raw_data
+        self.raw_requestline = raw_requestline
+        self.requestline = requestline
+        self.request_version = request_version
+
+    def __str__(self) -> str:
+        """
+        String representation of the first line of the request.
+        """
+        return self.requestline
+
+    @property
+    def content_type(self) -> Optional[str]:
+        """
+        Returns the request's content type
+
+        :return Content type header or None
+        """
+        if _CONTENT_TYPE_KEY in self.headers:
+            return self.headers[_CONTENT_TYPE_KEY]
+        else:
+            return None
+
+    @property
+    def json(self) -> Optional[dict]:
+        """
+        Returns the request's data in JSON format if exists
+
+        :return Request's data in JSON format or None
+        """
+        if (
+            self.raw_data is not None
+            and self.content_type == "application/json"
+        ):
+            return json.loads(self.raw_data)
+        else:
+            return None
+
+
+class HTTPResponse:
+    """
+    HTTPResponse containing the status code, headers and content.
+    """
```

Comment:
Looks like a namedtuple to me. Lets discuss.

Reply:
This code was already here, I only moved it upper because HTTPRequestHandler uses it.